[css-highlight-api] Export CSS highlights internal WPT

nt1m opened the pull request:
#### 13b4f972d55df5f75dcaed233cbc11e6e9bcd3bc
<pre>
[css-highlight-api] Export CSS highlights internal WPT
<a href="https://bugs.webkit.org/show_bug.cgi?id=277577">https://bugs.webkit.org/show_bug.cgi?id=277577</a>
<a href="https://rdar.apple.com/133098489">rdar://133098489</a>

Reviewed by Megan Gardner and Sam Sneddon.

Upstream all WPTs from LayoutTests/http/wpt/css/css-highlight-api, except the one that dumps repaint rects which is moved to fast/css.

* LayoutTests/TestExpectations:
* LayoutTests/fast/css/highlight-text-repaint-expected.txt: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-repaint-expected.txt.
* LayoutTests/fast/css/highlight-text-repaint.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-repaint.html.
* LayoutTests/http/wpt/css/css-highlight-api/highlight-text-expected.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-image-expected-mismatch.html: Copied from LayoutTests/http/wpt/css/css-highlight-api/highlight-image-expected-mismatch.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-image-notref.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-image-expected-mismatch.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-image.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-image.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-across-elements-expected.html: Copied from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-across-elements-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-across-elements-ref.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-across-elements-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-across-elements.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-across-elements.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-cascade-expected.html: Copied from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-cascade-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-cascade-ref.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-cascade-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-cascade.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-cascade.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-decorations-expected.html: Copied from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-decorations-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-decorations-ref.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-decorations-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-decorations.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-decorations.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-dynamic-expected-mismatch.html: Copied from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-dynamic-expected-mismatch.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-dynamic-notref.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-dynamic-expected-mismatch.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-dynamic.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-dynamic.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-replace-expected.html: Copied from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-replace-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-replace-ref.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-replace-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-replace.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-replace.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text.html.
* LayoutTests/imported/w3c/web-platform-tests/images/blank-highlight.png: Renamed from LayoutTests/http/wpt/images/blank-highlight.png.
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/ios/fast/css/highlight-text-repaint-expected.txt: Renamed from LayoutTests/platform/ios/http/wpt/css/css-highlight-api/highlight-text-repaint-expected.txt.

Canonical link: <a href="https://commits.webkit.org/281866@main">https://commits.webkit.org/281866@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3c38e9c737aa7b9e42ca2784528e83b47a689c8b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/61258 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/40619 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/13839 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/65208 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/11807 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/48297 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/12082 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/49504 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/8208 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/63292 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/37770 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/53068 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/30336 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/34448 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/10293 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/10720 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/56265 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/10588 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/66939 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/5205 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/10394 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/56877 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/5228 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/53031 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/57078 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/4301 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9213 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/36423 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/37506 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/38600 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/37250 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

gsnedders reviewed (approved):
rubber-stamping this, looks fine provided EWS passes

nt1m committed:
[css-highlight-api] Export CSS highlights internal WPT
https://bugs.webkit.org/show_bug.cgi?id=277577
rdar://133098489

Reviewed by Megan Gardner and Sam Sneddon.

Upstream all WPTs from LayoutTests/http/wpt/css/css-highlight-api, except the one that dumps repaint rects which is moved to fast/css.

* LayoutTests/TestExpectations:
* LayoutTests/fast/css/highlight-text-repaint-expected.txt: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-repaint-expected.txt.
* LayoutTests/fast/css/highlight-text-repaint.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-repaint.html.
* LayoutTests/http/wpt/css/css-highlight-api/highlight-text-expected.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-image-expected-mismatch.html: Copied from LayoutTests/http/wpt/css/css-highlight-api/highlight-image-expected-mismatch.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-image-notref.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-image-expected-mismatch.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-image.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-image.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-across-elements-expected.html: Copied from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-across-elements-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-across-elements-ref.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-across-elements-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-across-elements.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-across-elements.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-cascade-expected.html: Copied from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-cascade-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-cascade-ref.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-cascade-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-cascade.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-cascade.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-decorations-expected.html: Copied from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-decorations-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-decorations-ref.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-decorations-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-decorations.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-decorations.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-dynamic-expected-mismatch.html: Copied from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-dynamic-expected-mismatch.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-dynamic-notref.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-dynamic-expected-mismatch.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-dynamic.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-dynamic.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-replace-expected.html: Copied from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-replace-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-replace-ref.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-replace-expected.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text-replace.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text-replace.html.
* LayoutTests/imported/w3c/web-platform-tests/css/css-highlight-api/highlight-text.html: Renamed from LayoutTests/http/wpt/css/css-highlight-api/highlight-text.html.
* LayoutTests/imported/w3c/web-platform-tests/images/blank-highlight.png: Renamed from LayoutTests/http/wpt/images/blank-highlight.png.
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/ios/fast/css/highlight-text-repaint-expected.txt: Renamed from LayoutTests/platform/ios/http/wpt/css/css-highlight-api/highlight-text-repaint-expected.txt.

Canonical link: https://commits.webkit.org/281866@main
